frame/alliance/src/lib.rs

```diff
@@ -787,15 +800,38 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// As a member, give a retirement notice and start a retirement period required to pass in
+		/// order to retire.
+		#[pallet::weight(T::WeightInfo::retirement_notice())]
+		pub fn retirement_notice(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
Lexically, I think of dispatchable functions as verbs. "I want to {transfer, validate, join, nominate, vote, veto, give notice, etc.}."
```suggestion
		pub fn give_retirement_notice(origin: OriginFor<T>) -> DispatchResult {
```